testing/kuttl/e2e/security-context/00-assert.yaml

```diff
@@ -0,0 +1,180 @@
+apiVersion: postgres-operator.crunchydata.com/v1beta1
+kind: PostgresCluster
+metadata:
+  name: security-context
+status:
+  instances:
+    - name: instance1
+      readyReplicas: 1
+      replicas: 1
+      updatedReplicas: 1
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  labels:
+    postgres-operator.crunchydata.com/cluster: security-context
+    postgres-operator.crunchydata.com/pgbackrest-backup: replica-create
+status:
+  succeeded: 1
+---
+# initial pgBackRest backup
+apiVersion: v1
+kind: Pod
+metadata:
+  labels:
+    postgres-operator.crunchydata.com/cluster: security-context
+    postgres-operator.crunchydata.com/pgbackrest: ""
+    postgres-operator.crunchydata.com/pgbackrest-backup: replica-create
+    postgres-operator.crunchydata.com/pgbackrest-repo: repo1
+spec:
+  containers:
+  - name: pgbackrest
+    securityContext:
+      allowPrivilegeEscalation: false
+      privileged: false
+      readOnlyRootFilesystem: true
+      runAsNonRoot: true
+---
+# instance
+apiVersion: v1
+kind: Pod
+metadata:
+  labels:
+    postgres-operator.crunchydata.com/cluster: security-context
+    postgres-operator.crunchydata.com/data: postgres
+    postgres-operator.crunchydata.com/instance-set: instance1
+    postgres-operator.crunchydata.com/patroni: security-context-ha
+    postgres-operator.crunchydata.com/role: master
+spec:
+  containers:
```

Comment:
This looks like it will get all of the pods and containers but I wonder if there is a more generic way to do it. Could you have one assert that only checked for pods with the cluster label in metadata, then leave out the container name :recycle: ?
```
  containers:
  - securityContext:
      allowPrivilegeEscalation: false
      privileged: false
      readOnlyRootFilesystem: true
      runAsNonRoot: true
```

Reply:
I know that's more or less how the existing test does things, but I think there may be value in explicitly checking each container in case a container were to somehow be missing.

Otherwise, I'm a little unclear how best to do that with kuttl. I don't believe the above works as expected. I think we'd need to capture and loop through each container name, but I'm not aware with a good way to do that with kuttl. Any thoughts on that?

Reply:
I do not. I tried out a couple of things and I couldn't get it working. This seems like the way to go :+1: